File: Python/leetcode_561.py
from typing import List


def array_pair_sum(nums: List[int]) -> int:
    nums.sort()
    final_sum = 0
    for i in range(0, len(nums) - 1, 2):
        final_sum += min(nums[i], nums[i+1])
    return final_sum


print(array_pair_sum([1, 4, 3, 2]))

File: JavaScript/leetcode_561.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var arrayPairSum = function (nums) {
    nums.sort((a, b) => a - b);
    let sum = 0;
    for (let i = 0; i < nums.length; i += 2) {
        sum += Math.min(nums[i], nums[i + 1]);
    }
    return sum;
};

console.log(arrayPairSum([1, 4, 3, 2]));
